src/main/java/org/gridsuite/shortcircuit/server/utils/FeederResultSpecifications.java

```diff
@@ -0,0 +1,80 @@
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.gridsuite.shortcircuit.server.utils;
+
+import org.gridsuite.shortcircuit.server.dto.Filter;
+import org.gridsuite.shortcircuit.server.entities.FeederResultEntity;
+import org.gridsuite.shortcircuit.server.entities.ShortCircuitAnalysisResultEntity;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.util.List;
+import java.util.UUID;
+
+public final class FeederResultSpecifications {
+
+    // Utility class, so no constructor
+    private FeederResultSpecifications() {
+    }
+
+    public static Specification<FeederResultEntity> resultUuidEquals(UUID value) {
+        return (feederResult, cq, cb) -> cb.equal(feederResult.get("faultResult").get("result").get("resultUuid"), value);
+    }
+
+    public static Specification<FeederResultEntity> contains(String column, String value) {
+        return (feederResult, cq, cb) -> cb.like(feederResult.get(column), "%" + value + "%");
+    }
+
+    public static Specification<FeederResultEntity> startsWith(String column, String value) {
+        return (feederResult, cq, cb) -> cb.like(feederResult.get(column), value + "%");
+    }
+
+    public static Specification<FeederResultEntity> notEqual(String column, Double value) {
+        return (feederResult, cq, cb) -> cb.notEqual(feederResult.get(column), value);
+    }
+
+    public static Specification<FeederResultEntity> lessThanOrEqual(String column, Double value) {
+        return (feederResult, cq, cb) -> cb.lessThanOrEqualTo(feederResult.get(column), value);
+    }
+
+    public static Specification<FeederResultEntity> greaterThanOrEqual(String column, Double value) {
+        return (feederResult, cq, cb) -> cb.greaterThanOrEqualTo(feederResult.get(column), value);
+    }
+
+    public static Specification<FeederResultEntity> buildSpecification(ShortCircuitAnalysisResultEntity resultEntity, List<Filter> filters) {
+        Specification<FeederResultEntity> specification = Specification.where(resultUuidEquals(resultEntity.getResultUuid()));
+
+        if (filters == null || filters.isEmpty()) {
+            return specification;
+        }
+
+        for (Filter filter : filters) {
+            if (filter.dataType() == Filter.DataType.TEXT) {
+                String value = filter.value().toString();
+                switch (filter.type()) {
+                    case CONTAINS ->
+                            specification = specification.and(contains(filter.column(), value));
+                    case STARTS_WITH ->
+                            specification = specification.and(startsWith(filter.column(), value));
+                }
+            }
+            if (filter.dataType() == Filter.DataType.NUMBER) {
+                Double value = Double.valueOf(filter.value().toString());
```

Comment:
So if we reparse it, in filter we can have value as a String ? Otherwise don't parse it here ?

Reply:
I have to parse it from `String`, otherwise if we give a value `2` for instance, it will say it's an `integer` and not a `double` and the cast will fail...
I hesitated to use a `String` in the record directly.
But later on, maybe we can use more complitacted object (like Date) that we don't want to parse from String ?

Reply:
maybe use a different field name for different types ?
or if you really want it, maybe convert the int to double in the setter ?

Also maybe don't think too much about later one, follow the YAGNI principle